### ISSUE TITLE
bug: : exception during call chain: Docker not available when trying to create ec2 instance via terraform

### ISSUE THREAD

**Issue by techonelab** [localstack/localstack#7796]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

bug: : exception during call chain: Docker not available when trying to create ec2 instance via terraform
running localstack from docker

### Expected Behavior

given localstack is suppose to mock aws services, expected this to work

### How are you starting LocalStack?

With a `docker run` command

### Steps To Reproduce

#environment wsl ubuntu, installed docker manually not using docker desktop
#start localstack
docker run --rm -it -p 4566:4566 -p 4510-4559:4510-4559 -e LOCALSTACK_API_KEY=<> localstack/localstack-pro

was trying to walktrhough hcl's terraform(tutorial)  with modification to use localstack accordingly of course (change provider to point to localhost)
TLDR; just simply `terraform apply` 

other services seems to be working as expected but when I try to add ec2 instance creation with below code, I am getting the error "exception during call chain: Docker not available"

a part of my tf code used focused on instance...
---
resource "aws_instance" "web_server" {
  ami           = "ami-1e749f67"
  instance_type = "t2.micro"
  subnet_id     = aws_subnet.public_subnets["public_subnet_1"].id
  tags = {
    Name = "Ubuntu EC2 Server"
  }
}

### Environment

```markdown
- OS: windows 10, running WSL ubuntu 22.04
- LocalStack: latest pro
```

**Comment by techonelab** [comment]
thanks for this reminder, of course I forgot this one :)

**Comment by lukeduongdnanexus** [comment]
@thrau Hi I have this docker compose
`version: "3.8"

services:
  localstack:
    container_name: "${LOCALSTACK_DOCKER_NAME:-localstack-main}"
    image: localstack/localstack-pro:latest
    ports:
      - "127.0.0.1:4566:4566"            # LocalStack Gateway
      - "127.0.0.1:4510-4559:4510-4559"  # external services port range
    environment:
      # LocalStack configuration: https://docs.localstack.cloud/references/configuration/
      - DEBUG=${DEBUG:-0}
      - LOCALSTACK_AUTH_TOKEN=
    volumes:
      - "${LOCALSTACK_VOLUME_DIR:-./volume}:/var/lib/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"`
      
     but I have this bug
